src/fuse.rs

```diff
@@ -78,13 +79,16 @@ impl Filesystem for TreeFs {
         eprintln!("readdir - offset: {:?}", offset);
         match offset {
             0 => match self.0.get_by_id(ino) {
-                Some(Tree::Node { children, .. }) => {
+                Some(Tree {
+                    contents: Contents::Node(children),
+                    ..
+                }) => {
                     for child in children {
                         assert!(!reply.add(
-                            child.id(),
+                            child.id,
                             1,
                             TreeFs::tree_to_file_type(child),
-                            child.name()
+                            child.name.to_owned()
```

Comment:
```suggestion
                            &child.name
```
`add` is polymorphic in this argument and is fine with `&str`.